app/controllers/spree/adyen_notifications_controller.rb

```diff
@@ -6,14 +6,19 @@ class AdyenNotificationsController < StoreController
 
     def notify
       notification = AdyenNotification.build(params)
-      notification.save!
 
-      # prevent alteration to associated payment while we're handling the action
-      Spree::Adyen::NotificationProcessor.new(notification).process!
+      if notification.order.present?
+        process_order_notification!(notification)
+      else
+        process_notification!(notification)
+      end
+
       accept
-    rescue ActiveRecord::RecordNotUnique
+    rescue ActiveRecord::RecordNotUnique, ArgumentError
```

Comment:
`Spree::OrderMutex` throws an ArgumentError if no order is supplied. Perhaps being overly defensive here.